mood-analysis.rb

```diff
@@ -19,6 +19,52 @@ def analyze_mood(words)
   return ":-|"
 end
 
+def strip_punctuation(words)
+  words.downcase.gsub(/[^0-9^a-z ]/, '')
+end
+
+def happy_days(day)
+  count = 0
+  day_record = 0
+  num_days_before_happy_three = 0
+  text.each do |aline|
+    count += 1
+    line = strip_punctuation(aline)
+    face = analyze_mood(line)
+    (day_record += 1) if face==":-)"
+    if day_record == 3 && num_days_before_happy_three == 0
+      num_days_before_happy_three = count
+    end
+  end
+  if num_days_before_happy_three == 0
+    puts "WHY ARE YOU SO SAD"
+  else
+    puts "It takes #{num_days_before_happy_three} entries for 3 happy days to occur"
+  end
+end
```

Comment:
This method should return just a fixnum and then use that information in a puts instead of returning the string itself. That would make the information more versatile and able to be used in other situations rather than specific to this particular situation.